util.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import annotations
+from enum import Enum, auto
+import functools
+import logging
+from threading import Lock, Timer, Condition
+from typing import Callable
+
+import pigpio
+
+import influxdb_client
+from influxdb_client.client.write_api import SYNCHRONOUS
+from influxdb_client.rest import ApiException
+from urllib3.exceptions import NewConnectionError
+
+class SensorType(Enum):
+    TEMPERATURE = auto(),
+    HUMIDITY = auto(),
+    PRESSURE = auto(),
+    PM1 = auto(),
+    PM2_5 = auto(),
+    PM10 = auto()
+
+    @classmethod
+    @functools.cache
+    def index(cls, sensor_type: SensorType):
+        """@brief Returns index of element"""
+        return list(SensorType).index(sensor_type)
+
+    @classmethod
+    @functools.cache
+    def to_conf(cls, sensor_type: SensorType):
+        type_conf = {
+            SensorType.TEMPERATURE: "temperature_dht22_freq",
+            SensorType.HUMIDITY: "humidity_dht22_freq",
+            SensorType.PRESSURE: "pressure_bmp280_freq",
+            SensorType.PM1: "particle_pm1_pmsa003-c_freq",
+            SensorType.PM2_5: "particle_pm25_pmsa003-c_freq",
+            SensorType.PM10: "particle_pm10_pmsa003-c_freq",
+        }
+        return type_conf[sensor_type]
+
+
+class Key(Enum):
+    UP = auto()
+    DOWN = auto()
+    OK = auto()
+    CANCEL = auto()
+
+
+class Database:
```

Comment:
ABC and abstract methods?

Reply:
I made Database before InfluxDB and for testing purposes, frankly we could probably get rid of it and just use InfluxDatabase

Reply:
Sure, if it's redundant would be good idea to remove it.